packages/main/src/plugin/util/deferred.ts

```diff
@@ -32,3 +32,9 @@ export class Deferred<T> {
     });
   }
 }
+
+const AsyncFunction = Object.getPrototypeOf(async function () {}).constructor;
+
+export function isAsyncFunction(fn: unknown): boolean {
+  return fn?.constructor === AsyncFunction;
+}
```

Comment:
AFAIK the file is named deferred.ts and it contains Defered class, here it's strange to have such function in this file ?